### DISCUSSION TITLE
latest version of Ctrlr for Mac

### DISCUSSION THREAD

**Discussion by bijlevel** [RomanKubiak/ctrlr#202] [Q&A]
On the old forum site the latest for Mac is 5.3.198, but that version does not work with my panel developed in version 5.3.122. I did download from that same site version 6.0.26 for Windows, but again my panel didn't work in that version. So I was looking on this gitHub forum for the latest versions, but couldn't find them or only source code. So does one have to compile the code to get a working program? Or am I missing something?

**Comment by bijlevel** [comment]
Well, I will rephrase my question :-). How does one get the latest version of Ctrlr either for Windows or Mac, except by downloading the .exe or .dmg from the old forum?

**Reply by Godlike-Productions** [reply]
Click on Releases on the right hand side of the main github page, or go here https://github.com/RomanKubiak/ctrlr/releases.  Click the assets arrow under the 5.6.1 pre-release.  If you want anything further you'll need to build it yourself.

**Reply by bijlevel** [reply]
Thanks! Did download the version for Mac, but Ctrlr crashes :-( It shows a report and tells me that (a part of) it is send to a private location on gitHub, so I hope it gets the attention of mr. Kobiak.

**Reply by dnaldoog** [reply]
I would persevere and try and fit your panel to __5.3.198__ and on Windows __5.3.201__.

**Reply by bijlevel** [reply]
Thanks for the encouragement, but both versions you advised don't work on my computer, Mac and Windows 7 and 10. They do load my panel, but there is no MIDI activity, except for one button ( the only one of about 40 buttons) which does send/receive MIDI, which proves that there is a connection with the hardware. So at the end I have to conclude my panel does only work in the version I did develop it in (5.3.122 for Mac). I'm rather frustrated at the moment and considering quiting with Ctrlr in favour of Cycling '74 Max. I do have this program for many, many years, but did only a few projects with the software. I did upgrade two years ago to the latest version 8. I have good experiences with technical support and the forum. And also there are many good tutorials. Today I did explore the possibilities for making "panels" like Ctrlr and I did find several examples, with sysex an cc's, so it is doable. The downside is that I have to learn Javascript, because that's what MAX does support as scripting language and for sure I will need it to develop "panels". So for the time being I'm occupied with MAX -;)

**Reply by Ted-juh** [reply]
One out of 40 buttons, on all versions except the one you made the panel with? Strange.

Is the sysex/ midi correct when you fetch the data that's going out of Ctrlr with midi-ox or Bome Midi Translator?

**Reply by Godlike-Productions** [reply]
I think I know what's going on here. I'll ping you an email as well.

A quick summary.  The panel I uploaded to github had the LUA saved as files.  You'll either need to sync the Repo, or copy the files into the same directory structure manually.  I'll export what I've done as a .bpanelz file, which you can just open up.

**Reply by bijlevel** [reply]
I did try the panel Jason (Godlike-Productions) did send me, but same story. Not working (e.g. no MIDI activity except for one button) on Mac and Windows 7/10. @DnaldeT I use the MIDI monitor of Ctrlr and only the working button does show the sysex going out. The only difference of this button compared to the other 39 is that it does send the sysex from within Lua, while the other buttons send the sysex directly from the properties panel.

**Comment by dnaldoog** [comment]
Would it be possible to post the panel on this thread? I understand if you
don’t want to, but I feel like because every panel I have done on 5.3.201/5.3.198 has always
worked there must be something strange in there that can be fixed.

**Reply by dnaldoog** [reply]
Okay will have a look - you can delete it now if you like!

**Reply by bijlevel** [reply]
Thanks! I did delete the upload.

**Reply by dnaldoog** [reply]
Okay  - it seems like it will work on 5.3.201 - for some strange reason all those buttons are dead, but if I duplicate them with a new _uiButton_ the sysex is sent and the underlying callback functions work. The panel loads fine, at least on Windows 10.

**Reply by bijlevel** [reply]
Thanks John, the panel does now work in version 5.3.201 on Windows! I did send you a reply by email! I did also load the panel in version 5.3.198 for Mac and still the serious problem mentioned on the old forum (also by me, I believe) is present, namely "(no choices)" when trying to add a resource to a modulator, though the resource is actually added (and listed!) under the resource tab. That's why I will stick to version 5.3.122 for Mac.
